protocols/kad/CHANGELOG.md

```diff
@@ -16,10 +16,18 @@
   This would eventually lead to warning that says: "New inbound substream to PeerId exceeds inbound substream limit. No older substream waiting to be reused."
   See [PR 3152].
 
+- Refactor APIs to be streaming.
+  - Renamed `KademliaEvent::OutboundQueryCompleted` to `KademliaEvent::OutboundQueryProgressed`
+  - Instead of a single event `OutboundQueryCompleted`, there are now multiple events emitted, allowing the user to process them as they come in (via the new `OutboundQueryProgressed`).
+  - `get_providers` and `get_record` do not terminate automatically anymore, the caller needs to call `query.finish()`, on termination conditions.
```

Comment:
```suggestion
  - To finish a query early, i.e. before the final `OutboundQueryProgressed` of the query, a caller needs to call `query.finish()`.
```

They do terminate automatically. Though not before the final `OutboundQueryProgressed`. The above suggestion would make that easier to understand for me.